src/main/java/fi/aalto/cs/apluscourses/intellij/services/PluginSettings.java

```diff
@@ -62,12 +62,16 @@ private MainViewModel createNewMainViewModel(@NotNull Project project) {
     return new MainViewModel();
   }
 
-  public static boolean isShowReplConfigurationDialog() {
+  public static boolean shouldShowReplConfigurationDialog() {
     return propertiesManager.getBoolean(A_PLUS_SHOW_REPL_CONFIGURATION_DIALOG);
   }
 
-  //  todo consider to create a listener
-  public static void setShowReplConfigurationDialog(boolean showReplConfigDialog) {
+  /**
+   * Method to set property, responsible for showing REPL configuration window.
+   *
+   * @param showReplConfigDialog a {@link String} representing a boolean value of the flag.
+   */
+  public static void setShowReplConfigurationDialog(String showReplConfigDialog) {
```

Comment:
Is there a reason why we stick with IntelliJ's (somewhat awkward) "boolean stored as string" design solution? Could this method just take a boolean value and hide that conversion inside itself? Especially, since the getter returns boolean, it'd be nice if they were symmetrical.

Reply:
I've totally overlooked this. Fully agree with the idea of symmetry here, this is now fixed to `String`s.

I had the opposite idea here actually. AFAIK, the current approach of the platform for `boolean` properties provides only two options: `true` and **unset** (so the value is completely erased instead of setting it to `false` if 
```java 
setValue(false)
```
 is called. This would be acceptable if, say, they've used `Boolean` or some other non-dual-value-based structure. So my point is to **explicitly** emphasize (and propose for the future) the idea of having all the properties as `String`s.

Reply:
good idea, will do this and also translate some of the points from the comments on **nikke**'s open PR.

moreover, a little concern here... IDEA seems to write many configuration files like this one`build/idea-sandbox/config/options/other.xml` only after the first (the very first) run of the IDEA has **successfully** ended. In other words, `.xml`'s get created in some `onDestroyActivity` function (haven't checked where in particular). This means, that if the initial run of the IDE (simulate with `gradle clean` -> `gradle runIde` and just kill it from the host IDE), won't terminate properly, these files won't even be created.

Reply:
Nice observation! :+1: Certainly something that could confuse the hell out of me when debugging, so it's good to be aware of it.